azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -288,7 +279,7 @@ public VMImage getImage(final String id) {
 
    @Override
    public Iterable<Location> listLocations() {
-      final Iterable<String> whiteListZoneName = findWhiteListOfRegions();
+      final Iterable<String> whiteListedRegionNames = findWhiteListOfRegions();
```

Comment:
You are parsing this list in several places. Just call this in the constructor and assign to a class final variable.